pkg/mergemap/mergemap_test.go

```diff
@@ -0,0 +1,384 @@
+	}
+
+	dst := map[string]interface{}{
+		"data": map[string]interface{}{
+			"columns": []interface{}{
+				map[string]interface{}{
+					"name":        "calculation_id",
+					"data_type":   "STRING",
+					"description": "old description",
+					"is_nullable": true,
+				},
+				map[string]interface{}{
+					"name":        "entity_type",
+					"data_type":   "STRING",
+					"description": "",
+					"is_nullable": true,
+				},
+			},
+		},
+	}
+
+	src := map[string]interface{}{
+		"data": map[string]interface{}{
+			"columns": []interface{}{
+				map[string]interface{}{
+					"name":        "calculation_id",
+					"description": "updated description",
+				},
+				map[string]interface{}{
+					"name":        "signal_name",
+					"data_type":   "STRING",
+					"description": "new column",
+					"is_nullable": true,
+				},
+			},
+		},
+	}
+
+	result := Merge(dst, src, arrayMergeConfig)
+
+	data, ok := result["data"].(map[string]interface{})
+	if !ok {
+		t.Fatal("data field is not a map")
+	}
+
+	columns, ok := data["columns"].([]interface{})
+	if !ok {
+		t.Fatal("columns field is not an array")
+	}
+
+	if len(columns) != 3 {
+		t.Errorf("Expected 3 columns, got %d", len(columns))
+	}
+
+	calcColumn := columns[0].(map[string]interface{})
+	if calcColumn["name"] != "calculation_id" {
+		t.Errorf("Expected first column to be calculation_id, got %s", calcColumn["name"])
+	}
+	if calcColumn["description"] != "updated description" {
+		t.Errorf("Expected description to be updated, got %s", calcColumn["description"])
+	}
+	if calcColumn["data_type"] != "STRING" {
+		t.Errorf("Expected data_type to be preserved, got %s", calcColumn["data_type"])
+	}
+
+	entityColumn := columns[1].(map[string]interface{})
+	if entityColumn["name"] != "entity_type" {
+		t.Errorf("Expected second column to be entity_type, got %s", entityColumn["name"])
+	}
+
+	signalColumn := columns[2].(map[string]interface{})
+	if signalColumn["name"] != "signal_name" {
+		t.Errorf("Expected third column to be signal_name, got %s", signalColumn["name"])
+	}
+	if signalColumn["description"] != "new column" {
+		t.Errorf("Expected description to be 'new column', got %s", signalColumn["description"])
+	}
+}
+
+func TestMerge_ColumnsArrayPreservesOrder(t *testing.T) {
+	arrayMergeConfig := map[string]string{
+		"data.columns": "name",
+	}
+
+	dst := map[string]interface{}{
+		"data": map[string]interface{}{
+			"columns": []interface{}{
+				map[string]interface{}{"name": "col1"},
+				map[string]interface{}{"name": "col2"},
+				map[string]interface{}{"name": "col3"},
+			},
+		},
+	}
+
+	src := map[string]interface{}{
+		"data": map[string]interface{}{
+			"columns": []interface{}{
+				map[string]interface{}{"name": "col2", "updated": true},
+			},
+		},
+	}
+
+	result := Merge(dst, src, arrayMergeConfig)
+
+	columns := result["data"].(map[string]interface{})["columns"].([]interface{})
+
+	if len(columns) != 3 {
+		t.Errorf("Expected 3 columns, got %d", len(columns))
+	}
```

Comment:
this was we discussed before, special for array, if the table updated to has 3 cols, while previously it has 4 cols, then check by identifier, 1 cols should remove

if previously has 3 cols, and new has 4 cols, then check by identifier, the total should be 4 cols.

So, if your test case here `src` means new upadte, the result should be len(columns) == 1, since the new is less than before